src/gui/tag.rs

```diff
@@ -834,6 +834,22 @@ impl View for TagView {
             {
                 open_tag_in_default_application(self.tag_entry.reference.into());
             }
+
+            
+            if ui.button("Copy all hashes referencing this tag").clicked() {
+                let mut tag_hashes = Vec::new();
+                for (tag, entry) in &self.scan.references {
+                    tag_hashes.push(*tag);
+                }
+
+                let tag_hashes_str = tag_hashes
+                    .iter()
+                    .map(|hash| format!("{}", hash))
+                    .collect::<Vec<String>>()
+                    .join("\n");
```

Comment:
Intermediate array can be left out
```suggestion
                let tag_hashes_str = self.scan.references
                    .iter()
                    .map(|(hash, _entry)| format!("{}", hash))
                    .collect::<Vec<String>>()
                    .join("\n");
```